### PR TITLE
feat/test: updated issue wizard

### PR DESCRIPTION
**Pull request by sjoerdlmkns**

### CODE REVIEW

**Review by bobhageman — DISMISSED** [review]

File: integration_test/issue_wizard/ivido/ivido_demo_client.dart [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+import '../../helpers/helpers.dart';
+import '../../helpers/issuance_helpers.dart';
+import '../../irma_binding.dart';
+import '../../util.dart';
+
+Future<void> ividoDemoClientTest(WidgetTester tester, IntegrationTestIrmaBinding irmaBinding) async {
+  await pumpAndUnlockApp(tester, irmaBinding.repository);
+
+  // Navigate to the scanner screen
+  await tester.tapAndSettle(find.byKey(const Key('nav_button_scanner')));
```
Comment (ivard) [inline integration_test/issue_wizard/ivido/ivido_demo_client.dart:24]: In Firebase I once saw the issue that we cannot give camera permissions on iOS. This caused a permission notification to be present all the time. Is this still the case?

File: lib/src/models/issue_wizard.dart [hunk 2]
```diff
@@ -173,6 +173,7 @@ class IssueWizardItem {
   @JsonKey(name: 'inapp', defaultValue: false)
   final bool inApp;
 
+  @JsonKey(name: 'completed', defaultValue: false)
   final bool completed;
```
Comment (ivard) [inline lib/src/models/issue_wizard.dart:177]: I merged my PR already where all these `defaultValue` things are superfluous when they are exactly the same as the default value being defined in the constructor. When resolving the merge conflict, you might have to deal with this.

File: integration_test/issue_wizard/ivido/ivido_demo_client.dart (hunk 1 above)
Comment (sjoerdlmkns) [reply integration_test/issue_wizard/ivido/ivido_demo_client.dart:24]: In the test the permission dialog is shown but the test continues to run anyway. 

https://console.firebase.google.com/project/yivimobile/testlab/histories/bh.d812b7fd3ed364b4/matrices/6047455985802105590/executions/bs.eeabcecff89e561a/videos (12:00)

Do you think that this is an issue?

Comment (ivard) [reply integration_test/issue_wizard/ivido/ivido_demo_client.dart:24]: As long as the test is at the end, it's maybe fine. Previously, I disabled a test on iOS that had this same issue. https://github.com/privacybydesign/irmamobile/blob/ca5b7cf0ad5de830782aa164a39426c8a5cc271d/integration_test/home_test.dart#L94

That test was somewhere at the beginning, so then all tests would get the permission overlay.

File: assets/locales/nl.json [hunk 3]
```diff
@@ -371,7 +371,11 @@
     "done": "OK",
     "add": "Ophalen",
     "add_cards": "Gegevens ophalen",
-    "add_credential": "Haal {credential} op"
+    "add_credential": "Haal {credential} op",
+    "success": {
+      "header": "Gelukt",
+      "content": "Je hebt alle stappen successvol doorlopen"
```
Comment (sanderhollaar) [inline assets/locales/nl.json:377]: succes~s~vol

File: integration_test/issue_wizard/ivido/ivido_demo_client.dart [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+  // Test the questions and answers
+  final qAndAs = {
+    'Which data are added?':
+        'You retrieve your personal data. You need this data to show who you are. That way only you can log in on your account.',
+    'Where do you get the data?':
+        'You retrieve your personal data from the dutch personal records database ("Basis Registratie Personen", BRP). You log in with DigiD at the municipality of Nijmegen. Even if you live in another city. Nijmegen offers this service for everyone with a DigiD.',
+    'How does it work?':
+        'Your login data are stored only in your IRMA app on your phone. To log in at Ivido you show who you are with your IRMA app.',
+    'What is Ivido?':
+        'Ivido is a Personal Health Environment (PHE, dutch: PHO). In Ivido you can store everything about your healt. You choose yourself with who you share this data. You use IRMA to register and log in at Ivido.',
```
Comment (sanderhollaar) [inline integration_test/issue_wizard/ivido/ivido_demo_client.dart:66]: ~PHO~ PGO

File: integration_test/issue_wizard/ivido/ivido_demo_client.dart [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+  );
+
+  // Expect the right background color from Ivido
+  final headerWidget = headerFinder.evaluate().first.widget as IssuerVerifierHeader;
+  expect(headerWidget.backgroundColor, const Color(0xffe7dffe));
+
+  // Test the questions and answers
+  final qAndAs = {
+    'Which data are added?':
+        'You retrieve your personal data. You need this data to show who you are. That way only you can log in on your account.',
```
Comment (sanderhollaar) [inline integration_test/issue_wizard/ivido/ivido_demo_client.dart:60]: You'll retrieve your personal data. You'll use this data to log in to your account.

File: integration_test/issue_wizard/ivido/ivido_demo_client.dart [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+    matching: find.byType(IrmaCard),
+  );
+  expect(stepperCardsFinder, findsNWidgets(2));
+
+  // Expect the right content on the first card
+  expect(
+    tester.getAllText(stepperCardsFinder.first),
+    [
+      'Demo Personal data',
+      'Retrieve your personal data from the dutch personal records database ("Basis Registratie Personen", BRP). You do this with DigiD. You need this information to show who you are.',
```
Comment (sanderhollaar) [inline integration_test/issue_wizard/ivido/ivido_demo_client.dart:140]: **D**utch

File: integration_test/issue_wizard/ivido/ivido_demo_client.dart (hunk 4 above)
Comment (sjoerdlmkns) [reply integration_test/issue_wizard/ivido/ivido_demo_client.dart:66]: This is all part of the irma-demo-requestors scheme. 

https://github.com/privacybydesign/irma-demo-requestors/blob/3c39f28476ee579a29da008b003dc0e42fcc087d/requestors.json

Im just testing that what is described there.

File: integration_test/issue_wizard/ivido/ivido_demo_client.dart [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+    ],
+  );
+
+  // Expect the right content on the second card
+  expect(
+    tester.getAllText(stepperCardsFinder.at(1)),
+    [
+      'Demo Ivido Login',
+      'Retrieve your Ivido login pass. This way you can log at Ivido in easily and safely.',
```
Comment (sanderhollaar) [inline integration_test/issue_wizard/ivido/ivido_demo_client.dart:149]: ~pass. This way you can log at Ivido in easily and safely~ card to log in to Ivido.

File: integration_test/issue_wizard/ivido/ivido_demo_client.dart [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+  // Expect the right background color from Ivido
+  final headerWidget = headerFinder.evaluate().first.widget as IssuerVerifierHeader;
+  expect(headerWidget.backgroundColor, const Color(0xffe7dffe));
+
+  // Test the questions and answers
+  final qAndAs = {
+    'Which data are added?':
+        'You retrieve your personal data. You need this data to show who you are. That way only you can log in on your account.',
+    'Where do you get the data?':
+        'You retrieve your personal data from the dutch personal records database ("Basis Registratie Personen", BRP). You log in with DigiD at the municipality of Nijmegen. Even if you live in another city. Nijmegen offers this service for everyone with a DigiD.',
```
[the remark this answers is not in the document]
Comment (sjoerdlmkns) [reply integration_test/issue_wizard/ivido/ivido_demo_client.dart:62]: ?

File: lib/src/widgets/irma_card.dart [hunk 9]
```diff
@@ -90,6 +92,11 @@ class IrmaCard extends StatelessWidget {
         break;
     }
 
+    // Overwrite card color if a color is provided
```
Comment (sanderhollaar) [inline lib/src/widgets/irma_card.dart:95]: ~Overwrite~ Override

File: integration_test/issue_wizard/ivido/ivido_demo_client.dart [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+    'How does it work?':
+        'Your login data are stored only in your IRMA app on your phone. To log in at Ivido you show who you are with your IRMA app.',
```
Comment (sanderhollaar) [inline integration_test/issue_wizard/ivido/ivido_demo_client.dart:64]: log in ~at~ to Ivido

**Review by sanderhollaar — DISMISSED** [review]

**Review by sanderhollaar — APPROVED** [review]

**Review by ivard — APPROVED** [review]

**Pull request merged**